src/app/userDashboard/services/crud.service.ts

```diff
@@ -0,0 +1,78 @@
+import { Injectable, inject } from '@angular/core';
+import {
+  Firestore,
+  addDoc,
+  collection,
+  deleteDoc,
+  doc,
+  getDocs,
+} from '@angular/fire/firestore';
+import { ITrip } from '../models/trip';
+import { Observable, from } from 'rxjs';
+import { environment } from '../../../environments/environment';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class CrudService {
+  firestore = inject(Firestore);
+
+  addTrip(
+    userEmail: string,
+    tripName: string,
+    tripDestination: string
+  ): Observable<void> {
+    return from(
+      addDoc(
+        collection(
+          this.firestore,
+          `${environment.collectionNames.usersCollection}/${userEmail}/${environment.collectionNames.tripsCollection}`
```

Comment:
You reuse ` this.firestore,      ${environment.collectionNames.usersCollection}/${userEmail}/${environment.collectionNames.tripsCollection}` a lot, maybe just set it as a variable and reuse that variable. This will just clean up your code and make it more readable

Reply:
I can't make it a variable cause the userEmail comes in as a argument in the function and it's like in the middle of the path

Reply:
Sorry I see, I probably personally would have done ${environment.collectionNames.usersCollection} and ${environment.collectionNames.tripsCollection}, but that is just personal preference